rust/connlib/tunnel/src/client.rs

```diff
@@ -317,20 +265,13 @@ pub struct ClientState {
     /// DNS queries that we need to forward to the system resolver.
     buffered_dns_queries: VecDeque<DnsQuery<'static>>,
 
-    /// The (internal) IPs we have assigned for a certain DNS resource.
-    dns_resources_internal_ips: HashMap<IpAddr, DnsResource>,
-    /// DNS queries we can only answer once we have connected to the resource.
-    ///
-    /// See [`dns::ResolveStrategy`] for details.
-    deferred_dns_queries: HashMap<(DnsResource, Rtype), IpPacket<'static>>,
-    /// Hands out IPs for DNS resources.
-    ip_provider: IpProvider,
     /// Maps from connlib-assigned IP of a DNS server back to the originally configured system DNS resolver.
     dns_mapping: BiMap<IpAddr, DnsServer>,
     /// When to next refresh DNS resources.
     ///
     /// "Refreshing" DNS resources means triggering a new DNS lookup for this domain on the gateway.
     next_dns_refresh: Option<Instant>,
+    dns_map: DnsMap,
```

Comment:
I'd prefer if we can adopt a naming here that follows what DNS itself uses. Technically speaking, I think we are building an "(authoritative) name server" here?

Reply:
`stub_resolver` might be a better name

Reply:
Yeah true!